RecoTauTag/RecoTau/src/PFRecoTauClusterVariables.cc

```diff
@@ -235,6 +235,113 @@ namespace reco {
+
+          float sigCands_pt = 0.;
+          float sigCands_dr, sigCands_deta, sigCands_dphi;
+          sigCands_dr = sigCands_deta = sigCands_dphi = 0.;
+          for (auto& j : tau.signalGammaCands()) {
+            const float dr = reco::deltaR(tau, *j);
+            const float deta = std::abs(tau.eta() - j->eta());
+            const float dphi = std::abs(reco::deltaPhi(tau.phi(), j->phi()));
+            const float pt_ = j->pt();
+            sigCands_dr += dr * pt_;
+            sigCands_deta += deta * pt_;
+            sigCands_dphi += dphi * pt_;
+            sigCands_pt += pt_;
+          }
+          if (sigCands_pt > 0.) {
+            sigCands_dr = sigCands_dr / sigCands_pt;
+            sigCands_deta = sigCands_deta / sigCands_pt;
+            sigCands_dphi = sigCands_dphi / sigCands_pt;
+          } else {
+            sigCands_dr = sigCands_deta = sigCands_dphi = -0.1;
+          }
+          float isoCands_pt = 0.;
+          float isoCands_dr, isoCands_deta, isoCands_dphi;
+          isoCands_dr = isoCands_deta = isoCands_dphi = 0.;
+          for (auto& j : tau.isolationGammaCands()) {
+            const float dr = reco::deltaR(tau, *j);
+            const float deta = std::abs(tau.eta() - j->eta());
+            const float dphi = std::abs(reco::deltaPhi(tau.phi(), j->phi()));
+            const float pt_ = j->pt();
+            isoCands_dr += dr * pt_;
+            isoCands_deta += deta * pt_;
+            isoCands_dphi += dphi * pt_;
+            isoCands_pt += pt_;
+          }
+          if (isoCands_pt > 0.) {
+            isoCands_dr = isoCands_dr / isoCands_pt;
+            isoCands_deta = isoCands_deta / isoCands_pt;
+            isoCands_dphi = isoCands_dphi / isoCands_pt;
+          } else {
+            isoCands_dr = isoCands_deta = isoCands_dphi = -0.1;
+          }
+          mvaInput[9] = isoCands_deta;
+          mvaInput[10] = isoCands_dphi;
+          mvaInput[11] = isoCands_dr;
+          mvaInput[12] = sigCands_deta;
+          mvaInput[13] = sigCands_dphi;
+          mvaInput[14] = sigCands_dr;
+
+          float e = tau.hcalEnergy() + tau.ecalEnergy();
+          e > 0. ? e = tau.ecalEnergy() / e : e = -1.;
+          mvaInput[15] = e;
+          mvaInput[16] = tau.dxy() >= 0. ? +1 : -1;
+          mvaInput[17] = sqrt(std::abs(tau.dxy()));
+          mvaInput[18] = std::abs(tau.dxy_Sig());
+          mvaInput[19] = tau.ip3d() >= 0. ? +1 : -1;
+          mvaInput[20] = sqrt(std::abs(tau.ip3d()));
+          mvaInput[21] = std::abs(tau.ip3d_Sig());
+          mvaInput[22] = tau.hasSecondaryVertex();
+          mvaInput[23] = decayDistMag;  //sqrt(tau.flightLength().Mag2());
+          mvaInput[24] = tau.flightLengthSig();
+          mvaInput[25] = leadingTrackChi2;  //tau.leadingTrackNormChi2();
+
+          float thetaGJmax, thetaGJ;
+          thetaGJmax = thetaGJ = 0.;
+          if (decayDistMag > 0.) {
+            const float mTau = 1.77686;
```

Comment:
define this constant in a more generic location (it does not depend on any conditional statement)